Fix AdSense stepper styles

tofumatt opened the pull request:
## Summary

<!-- Please reference the issue this PR addresses in the following list. -->
Addresses issue:

- #7303

## Relevant technical choices

Had to make a few more tweaks than were mentioned in the IB to get both the mobile and desktop views to work but I think this does the trick 🙂 

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

nfmohit commented on assets/sass/components/adsense/_googlesitekit-adsense-ad-blocking-recovery.scss:
```diff
@@ -82,11 +82,14 @@
 
 	.googlesitekit-ad-blocking-recovery__footer-cancel {
 		border-top: 1px solid $c-mercury;
-		padding: 17px;
+		padding-top: 14px;
 
 		@media (min-width: $bp-desktop) {
 			border: none;
-			padding: 0;
+		}
+
```
![image](https://github.com/google/site-kit-wp/assets/20284937/0e42374e-9a0f-4ff6-9f27-a333bf5cb0f3)

It looks like the `Cancel` action is slightly misaligned vertically.

![image](https://github.com/google/site-kit-wp/assets/20284937/ce9767ec-a218-4437-8648-244a1336e015)

Also, in smaller viewports, the border doesn't cover the entire width of the container.

Could we remove the border and padding entirely from `.googlesitekit-ad-blocking-recovery__footer-cancel` and let `.googlesitekit-ad-blocking-recovery__footer` handle that instead?

tofumatt replied:
Done, way better, thanks 🙏🏻

nfmohit commented on assets/sass/components/adsense/_googlesitekit-adsense-ad-blocking-recovery.scss:
```diff
@@ -58,13 +58,13 @@
 		display: flex;
 		flex-direction: column;
 		justify-content: space-between;
+		padding: 14px $grid-gap-desktop;
 
 		@media (min-width: $bp-desktop) {
 			align-items: center;
 			border-top: 1px solid $c-mercury;
 			flex-direction: row;
 			margin: 42px 0 0;
```
Since the footer only contains one `Cancel` action, I don't think these flex attributes are needed. They may have been left over after copying the styles from the User Input V2 screen.

Following up on the last comment, removing the media query here, and applying only the `padding`, `border-top`, and `margin` styles to `.googlesitekit-ad-blocking-recovery__footer` seems to resolve the problem.

tofumatt replied:
That works great, thanks! 👍🏻

nfmohit commented on assets/sass/components/global/_googlesitekit-stepper.scss:
```diff
@@ -74,13 +82,22 @@
 
 	.googlesitekit-stepper__step-title--upcoming {
 		color: $c-surfaces-on-surface-variant;
+		display: block;
+
+		@media (min-width: $bp-tablet) {
+			display: flex;
+		}
```
Could you help me understand the purpose of this change on this specific variant? Is it because of the string length? If so, do you think it'd be wiser to apply it on `.googlesitekit-stepper__step-title` itself instead of a specific variant?

We could make `.googlesitekit-stepper__step-title` use a `display` `block` and an unset `height` by default, but apply a `display` `flex` and `height` `24px` starting from `$bp-tablet`.

On mobile viewports, if the spacing between the step number and the title is an issue, we could reduce the margin in that case.

What do you think?

tofumatt replied:
I think this was to fix something after another change I made, but with your suggested tweaks is no longer needed 😆

nfmohit reviewed (changes requested):
Hi, @tofumatt, great work here!

I have left a few comments for your attention, after addressing which, this should be good to go.

The VRTs are failing for some reason and I do not see any noticeable difference between the references and the tests, but I think they could get solved if the suggested changes are considered.

Let me know what you think, thank you!

tofumatt commented:
Thanks a bunch for the suggestions in the review @nfmohit! They all work a lot better and remove the tweaks I made for the `display: flex` as well; I honestly forget why I needed the fix but after some adjustments it isn't needed 🙂

tofumatt commented:
Hmm, oddly I updated these tests locally and they pass, but they seem to be failing here. I've copied over the failing tests from CI and updated the branch, hopefully it's fine now 🤔

nfmohit reviewed (approved):
LGTM, the VRTs are passing now as well, thank you @tofumatt!
